source/extensions/filters/http/oauth2/filter.cc

```diff
@@ -507,6 +540,48 @@ void OAuth2Filter::finishGetAccessTokenFlow() {
   config_->stats().oauth_success_.inc();
 }
 
+void OAuth2Filter::finishUpdateAccessTokenFlow() {
```

Comment:
nit: use `finishRefreshAccessTokenFlow` because the `onRefreshAccessTokenSuccess`. I think it would be better to unify these name to use `Update` or `Refresh`.